spec.md

```diff
@@ -144,18 +144,57 @@ circuit MyTop...
 
 ## Circuits
 
-All FIRRTL circuits consist of a list of modules, each representing a hardware
-block that can be instantiated. The circuit must specify the name of the
-top-level module.
+A FIRRTL circuit is a named collection of FIRRTL modules.  Each module is a
+hardware "unit" that has ports, registers, wires, and may instantiate other
+modules (see: [@sec:modules]).  (This is the same concept as a Verilog
+`module`{.verilog}.)  A public module is usable outside of the current circuit.
```

Comment:
Perhaps "Public modules may be instantiated outside the circuit".  This defined "usable" more narrowly.

Further, something about public modules being the exported identifiers for a circuit and all other module kinds not being namable outside the circuit might help clarify.